pkg/repository/manifest/manifest.go

```diff
@@ -222,16 +248,45 @@ func readTimestampManifest(input io.Reader, keys crypto.KeyStore) (*Timestamp, e
 	return &ts, nil
 }
 
+// SignAndWrite creates a manifest and writes it to out.
+func SignAndWrite(out io.Writer, role ValidManifest, keyID string, privKey crypto.PrivKey) error {
```

Comment:
Seems we can remove the `keyID` parameter if we can obtain the fingerprint from public key.